A3.md

```diff
@@ -165,21 +175,27 @@ message TraceNode {
 message SubchannelTrace {
   // tracks total trace nodes seen, since many may be overwritten
   int32 num_nodes_logged = 1;
+  // time this subchannel tracer was created
+  Timestamp time = 2;
   // all of the nodes for this subchannel
-  repeated TraceNode nodes = 2;
+  repeated TraceNode nodes = 3;
 }
 
 // trace for one channel
 message ChannelTrace {
   // tracks total subchannel seen, since some may be dropped as they go inactive
   int32 num_subchannels_logged = 1;
+  // all of the nodes for this channel
+  // time this channel tracer was created
+  Timestamp time = 2;
+  repeated TraceNode nodes = 3;
   // all of the subchannels for this channel
-  repeated SubchannelTrace subchannels = 2;
+  repeated SubchannelTrace subchannels = 4;
 };
 ```
 
 ### Garbage Collection
 
-In order to ensure that the tracer(s) do not use too much memory, both trace nodes and subchannel tracers will be discarded after a time. All trace nodes will be stored in circular buffers, the size of which can be set by toggling the channel tracer channel arg.
+In order to ensure that the tracer(s) do not use too much memory, both trace nodes and subchannel tracers will be discarded after a time. All tracers will only hold up to a hard limit of trace nodes, the size of which can be set by toggling a channel argument.
```

Comment:
"after a time" -> "when the number of trace nodes has gotten too high"
"hard limit" -> "configurable maximum number"
"size of which" -> "value of which"
"toggling" -> "setting"